pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -284,11 +288,11 @@ func (c *ruleCache) PatchAppliedToGroup(patch *v1beta1.AppliedToGroupPatch) erro
 	if !exists {
 		return fmt.Errorf("AppliedToGroup %v doesn't exist in cache, can't be patched", patch.Name)
 	}
-	for _, added := range patch.AddedPods {
-		podSet.Insert(&added)
+	for i := range patch.AddedPods {
+		podSet.Insert(&patch.AddedPods[i])
 	}
-	for _, removed := range patch.RemovedPods {
-		podSet.Delete(&removed)
+	for i := range patch.RemovedPods {
+		podSet.Delete(&patch.RemovedPods[i])
```

Comment:
do you think we need a similar change on L356 in cache.go?

Reply:
Thanks for reminding. I checked L356 doesn't store the address of the loop iterator variable so should be fine. But I still changed it to get item address via index just in case that `toRule` keeps the passed address in future.

Reply:
Thanks